partcad/src/partcad/assembly.py

```
@@ -64,67 +58,63 @@ def add(
-            self.shape = compound.wrapped
-        if self.shape is None:
-            pc_logging.warning("The shape is None")
-        return self.shape
+        if "child" not in self.config:
+            # This is the top level assembly
+            with pc_logging.Action("Assembly", self.project_name, self.name):
+                return await self._get_shape_real(ctx)
+        else:
+            return await self._get_shape_real(ctx)
+
+    async def _get_shape_real(self, ctx):
```

Ruff (0.8.2)

Missing return type annotation for private function `_get_shape_real`

(ANN202)